Gemfile

```diff
@@ -1,5 +1,11 @@
-source "http://rubygems.org"
+source 'https://rubygems.org'
+ruby '~> 2.6.5'
 
 gemspec
 
-gem 'activerecord-redshift-adapter', :git => 'https://github.com/arp/activerecord-redshift-adapter', :branch => 'rails4-compatibility'
+gem 'bulk_data_methods', '~> 1.1.0'
+gem 'jquery-rails', '~> 4.3.0'
+gem 'pg', '~> 1.2.0'
+gem 'rails', '~> 5.0.0'
+gem 'rspec-rails', '~> 4.0.0'
+gem 'sprockets', '~> 4.0'
```

Comment:
Should we have these dependencies in the gemspec file?

Reply:
They already are.

They're defined as loose dependencies in the gemspec, which allows ranges for both Rails 4 and Rails 5.

```rb
s.add_dependency 'rails', `>= 4.0.0`
```

Tighter, version-specific dependencies are listed on the Gemfiles:

```rb
# Gemfile
gem 'rails', '~> 5.0.0'

# Gemfile-legacy
gem 'rails', '~> 4.0.0'
```

Other gems follow the same principle.

| Gem | Rails 4 | Rails 5 |
| -- | -- | -- |
| pg | 0.18.x | 1.x | 
| rspec-rails | 3.x | 4.x |
| sprockets | 3.x | 4.x |

Reply:
I understand you have Gemfile-rails4 and Gemfile-rails5, but is Gemfile also necessary? It might be a little confusing to see many files for dependencies.

Reply:
> I understand you have Gemfile-rails4 and Gemfile-rails5, but is Gemfile also necessary?

There is no `Gemfile-rails5`. There are only two Gemfiles:

- `Gemfile` &mdash; the primary-supported environment (Rails 5), and
- `Gemfile-rails4` &mdash; the legacy environment.